File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int count=0;
        int ans=-1;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(count==0){
                ans=nums[i];
                count++;
            }
            else if(nums[i]==ans)
                count++;
            else
                count--;
        }
        return ans;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ele=nums[0];
        int count=1;
        for(int i:nums){
            if(i==ele)
                count++;
            else{
                if(i!=ele)
                    count--;
                if(count==0){
                    ele=i;
                    count=1;
                }
            }
        }
        return ele;
    }
}